tests/unit/test_models.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+from fauna import Document, NamedDocument, Module, DocumentReference, NamedDocumentReference
+
+
+def test_document_required_props(subtests):
+    with subtests.test(msg="must contain 'coll'"):
+        with pytest.raises(ValueError,
+                           match="Data must contain the 'coll' and 'id' keys"):
+            Document({"id": 123})
+
+    with subtests.test(msg="must contain 'id'"):
+        with pytest.raises(ValueError,
+                           match="Data must contain the 'coll' and 'id' keys"):
+            Document({"coll": "hi"})
+
+
+def test_named_document_required_props(subtests):
+    with subtests.test(msg="must contain 'coll'"):
+        with pytest.raises(
+                ValueError,
+                match="Data must contain the 'coll' and 'name' keys"):
+            NamedDocument({"name": 123})
+
+    with subtests.test(msg="must contain 'id'"):
+        with pytest.raises(
+                ValueError,
+                match="Data must contain the 'coll' and 'name' keys"):
+            NamedDocument({"coll": "hi"})
+
+
+def test_ref_does_not_conflict(subtests):
+    with subtests.test(msg="Document does not conflict"):
+        d = Document({"id": 123, "coll": Module("Dogs"), "ref": "my_ref"})
```

Comment:
Documents from fauna will store `id` as a numeric string, so this should be
```suggestion
        d = Document({"id": "123", "coll": Module("Dogs"), "ref": "my_ref"})
```